From c6a3d8e274f79d1e62142e1636c3424e38093f90 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 1 Mar 2017 17:16:37 +0000
Subject: [PATCH 1/3] conda: Unbundle libxgboost.{so,dll}

.. and put libxgboost.so into $PREFIX/lib/
.. and libxgboost.dll into $PREFIX/Library/mingw-w64/lib/
---
 python-package/setup.py           | 8 +++++++-
 python-package/xgboost/libpath.py | 4 ++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/python-package/setup.py b/python-package/setup.py
index e6c3386f..79c4ad2e 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -17,6 +17,12 @@ libpath = {'__file__': libpath_py}
 exec(compile(open(libpath_py, "rb").read(), libpath_py, 'exec'), libpath, libpath)
 
 LIB_PATH = [os.path.relpath(libfile, CURRENT_DIR) for libfile in libpath['find_lib_path']()]
+if os.getenv('CONDA_BUILD'):
+    print("Conda-build detected, not installing libxgboost from: %s" % LIB_PATH)
+    data_files=('xgboost')
+else:
+    print("Install libxgboost from: %s" % LIB_PATH)
+    data_files=('xgboost', LIB_PATH)
 print("Install libxgboost from: %s" % LIB_PATH)
 # Please use setup_pip.py for generating and deploying pip installation
 # detailed instruction in setup_pip.py
@@ -35,7 +41,7 @@ setup(name='xgboost',
       # this will use MANIFEST.in during install where we specify additional files,
       # this is the golden line
       include_package_data=True,
-      data_files=[('xgboost', LIB_PATH)],
+      data_files=[data_files],
       license='Apache-2.0',
       classifiers=['License :: OSI Approved :: Apache Software License'],
       url='https://github.com/dmlc/xgboost')
diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
index 6abb9ad7..fa8406ae 100644
--- a/python-package/xgboost/libpath.py
+++ b/python-package/xgboost/libpath.py
@@ -24,6 +24,10 @@ def find_lib_path():
     dll_path = [curr_path, os.path.join(curr_path, '../../lib/'),
                 os.path.join(curr_path, './lib/'),
                 os.path.join(sys.prefix, 'xgboost')]
+    if sys.platform == 'win32':
+        dll_path = [os.path.join(sys.prefix, 'Library', 'mingw-w64', 'bin']
+    else:
+        dll_path = [os.path.join(sys.prefix, 'lib', ')]
     if sys.platform == 'win32':
         if platform.architecture()[0] == '64bit':
             dll_path.append(os.path.join(curr_path, '../../windows/x64/Release/'))
-- 
2.14.3 (Apple Git-98)

